lib/ansible/modules/system/group.py

```diff
@@ -120,9 +120,14 @@ def group_del(self):
         cmd = [self.module.get_bin_path(command_name, True), self.name]
         return self.execute_command(cmd)
 
+    def _local_check_gid_exists(self):
+        if self.gid and self.gid in [gr.gr_gid for gr in grp.getgrall()]:
```

Comment:
It may be slightly faster to only call `gr.getgrall()` once. I'm not sure how expensive it is, espceially on a system using directory authentication.
```suggestion
        all_groups = grp.getgrall()
        if self.gid and self.gid in [gr.gr_gid for gr in all_groups]:
```

Reply:
I was incorrect about this. The expression is evaluated once, but the a _conditional_ in the expression is evaluated on each iteration.